stew/byteutils.nim

```diff
@@ -46,7 +46,11 @@ func hexToByteArrayImpl(hexStr: openArray[char], output: var openArray[byte], fr
   ## Allows specifying the byte range to process into the array
   var sIdx = skip0xPrefix(hexStr)
 
+  if output.len < 1:
```

Comment:
why are you checking the output length here?

also, even when using the original code without this check, your new test passes, so this change effectively does not make a difference.

Reply:
Oh I'm sorry I made this change in haste!

Reply:
I had a problem because I was on my `feature/execution_api_spec` branch which was checked out while there was error handling in my `web3` impl.

Reply:
How does it look now?

Reply:
well, as I mentioned above, your new test case triggers a value error already without any code changes, so the bug you're looking for is elsewhere - this function works as expected, when given "0x" as string